app/views.py

```diff
@@ -1,16 +1,33 @@
 from flask import session, redirect, url_for, request, render_template, flash
 from app import app
-from datetime import timedelta
+from datetime import datetime
 import pymesync
 import forms
 import re
 
 
-# Expires users after 30 minutes OF UNACTIVITY
-@app.before_request
-def make_session_permanent():
-    session.permanent = True
-    app.permanent_session_lifetime = timedelta(minutes=30)
+def isLoggedIn():
+    """Checks if the user is logged in. Also checks token expiration time,
+        logging the user out if their token is expired."""
+
+    if 'token' not in session:
+        return False
+
+    ts = pymesync.TimeSync(baseurl=app.config['TIMESYNC_URL'],
+                           test=app.config['TESTING'],
+                           token=session['token'])
+
+    expire = ts.token_expiration_time()
+
+    if type(expire) is dict and ('error' in expire or
```

Comment:
ugh. Would it be easier to handle if pymesync just threw an error instead of returning one in a dict? I don't like that you have to check the return type here. Returning an error message in a dict makes more sense for methods that return a dict.

@Kennric @subnomo